File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    visited = [0] * n     
    # di = [1,-1,0,0]
    # dj = [0,0,1,-1]
    
    def bfs(i):
        q = deque()
        q.append(i)
        visited[i] = 1
        
        while q:
            y = q.popleft()
            
            for c in range(n):
                if computers[y][c] == 1 and visited[c] == 0:
                    visited[c] = 1
                    q.append(c)
        
        return
        
    
    for i in range(n):
        if visited[i] == 0:
            answer +=1
            bfs(i)
            
            
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    bool visited[200] = {false,};
    queue<int>q;
    for(int i=0;i<n;i++){
        if(visited[i]) continue;
        q.push(i);
        visited[i] = true;
        while(!q.empty()){
            int now = q.front();
            q.pop();
            for(int j = 0;j<computers[now].size();j++){
                if(computers[now][j] != 1) continue;
                if(visited[j]) continue;
                q.push(j);
                visited[j] = true;
            }
        }
        answer++;
    }
    return answer;
}